Add `ballerinax/weaviate` connector (#754)
ballerina-platform/openapi-connectors: openapi/weaviate/Package.md

Connects to the [Weaviate Vector Search Engine API](https://weaviate.io/developers/weaviate/api) from Ballerina.

## Package overview

> The `ballerinax/weaviate` package is a [Ballerina](https://ballerina.io/) connector for Weaviate Vector Search Engine API. The connector provide access to the manipulations of weaviate schema, objects and search vectors based on various criterias. 

Reviewer: ```suggestion
The `ballerinax/weaviate` package is a [Ballerina](https://ballerina.io/) connector for the Weaviate Vector Search Engine API. The connector provides access to the manipulations of the Weaviate schema, objects, and search vectors based on various criteria. 
```

Reviewer 2: Addressed in https://github.com/ballerina-platform/openapi-connectors/pull/754/commits/4011e12c5729754ff4de72e28289bb1362275fd0.

### Compatibility
|                            | Version                       |
|----------------------------|-------------------------------|

> | Ballerina Language         | 2201.4.1 (Swan Lake Update 4) |

Reviewer: ```suggestion
| Ballerina language         | 2201.4.1 (Swan Lake Update 4) |
```
Shall we give the link to the Download page here?

Reviewer 2: Addressed in https://github.com/ballerina-platform/openapi-connectors/pull/754/commits/4011e12c5729754ff4de72e28289bb1362275fd0.

> | Ballerina Language         | 2201.4.1 (Swan Lake Update 4) |

Reviewer 3: Suggested by Jo because Ballerina Central can automatically provide the compatible versions. Even when new releases are available. We can get other's opinion on this too.

> | Ballerina Language         | 2201.4.1 (Swan Lake Update 4) |

Reviewer 2: Let's wait until we get the feedback from Ballerina team since I believe there should be consistency among all connectors we have.

> | Ballerina Language         | 2201.4.1 (Swan Lake Update 4) |

Reviewer 4: Looping in @abeykoon. Could you please share the recommended `module.md`, `pacakge.md` and `README.md` templates for ballerinax packages?

> | Weaviate Vector Search API | 1.18.0                        |

Reviewer: ```suggestion
| Weaviate vector eearch API | 1.18.0                        |
```

Reviewer 2: Addressed in https://github.com/ballerina-platform/openapi-connectors/pull/754/commits/4011e12c5729754ff4de72e28289bb1362275fd0.

> 

Reviewer 2: Please refer the Package.md template provided in the checklist. https://github.com/ballerina-platform/ballerina-extended-library/discussions/77.

> 

Reviewer 3: This change is suggested by Jo and I think it is better to do that improvement eventhough it's not yet updated in the PR checklist. Anyway we can get other's opinion also.

> 

Reviewer 2: Let's wait until we get the feedback from Ballerina team since I believe there should be consistency among all connectors we have.

> 

Reviewer 4: Looping in @abeykoon. Could you please share the recommended `module.md`, `pacakge.md` and `README.md` templates for ballerinax packages?
